MobileWallet/TariLib/Wrappers/Wallet.swift

```diff
@@ -358,23 +359,40 @@ class Wallet {
         let mutable = UnsafeMutablePointer<Int8>(mutating: resultPtr!)
         string_destroy(mutable)
 
-        return result
+        let (walletPubKey, publicKeyError) = self.publicKey
+        guard publicKeyError == nil else {
+            throw publicKeyError!
+        }
+
+        let splitResult = result.components(separatedBy: "|")
+        guard splitResult.count == 2 else {
+            throw WalletErrors.invalidSignatureAndNonceString
+        }
+
+        return Signature(hex: splitResult[0], nonce: splitResult[1], message: message, publicKey: walletPubKey!)
```

Comment:
same here. probably you checked for error. there is no error then it's safe to use !

Reply:
Yes same with TariLib if the error is nil, the result will never be nil